WidgetExtension/Up Next/EpisodeView.swift

```diff
@@ -62,27 +55,57 @@ struct EpisodeView: View {
     }
 }
 
-struct WidgetPlayToggleStyle: ToggleStyle {
-    func makeBody(configuration: Configuration) -> some View {
-        ZStack {
-            configuration.label
-                .truncationMode(.tail)
+struct WidgetFirstEpisodePlayToggleStyle: ToggleStyle {
+    let colorScheme: PCWidgetColorScheme
 
-            Circle()
-                .foregroundStyle(.white)
-                .frame(width: 24, height: 24)
+    func makeBody(configuration: Configuration) -> some View {
+        HStack(spacing: 0) {
             Group {
                 configuration.isOn ?
                 Image("icon-pause")
                     .resizable()
-                    .foregroundStyle(.black)
+                    .foregroundStyle(colorScheme.topButtonTextColor)
                 :
-                    Image("icon-play")
+                Image("icon-play")
                     .resizable()
-                    .foregroundStyle(.black)
+                    .foregroundStyle(colorScheme.topButtonTextColor)
+            }
+            .frame(width: 28, height: 28)
+            // TODO: Something fun - create a timeline that counts down by the minute instead of showing "now playing"
+            configuration.label
+                .truncationMode(.tail)
+        }
+        .padding(.trailing, 12) // icon has 8px padding built into it, so this should match 8 + .leading
+        .padding(.leading, 4)
+        .padding(.vertical, 2) // 2 + 8 (from icon) = 10 in design (actually 9.76)
+        .background(
+            RoundedRectangle(cornerRadius: 100)
+                .foregroundColor(colorScheme.topButtonBackgroundColor)
+        )
+    }
+}
+
+struct WidgetPlayToggleStyle: ToggleStyle {
+    let colorScheme: PCWidgetColorScheme
+
+    func makeBody(configuration: Configuration) -> some View {
```

Comment:
Should this be marked `@ViewBuilder` in case the view is not a single View? I think the protocol method is marked ViewBuilder, so maybe that's automatically applied?

Reply:
I think its fine to not mark it because as you said the protocol method is marked `@ViewBuilder` (among other things). The sample code in the documentation also doesn't specify it https://developer.apple.com/documentation/swiftui/togglestyle/makebody(configuration:)